### PR TITLE
Force check dependencies for provides and requires

### PR DESCRIPTION
**Pull request by aandvalenzuela**
Testing with `oracle` pkg. `AutoReq` to be removed also in [‎`ucx`](https://github.com/aandvalenzuela/cmsdist/blob/a99b9b044d3ca31f8e5c3e3977c6c0c5414b3a53/ucx.spec#L11), [`llvm`](https://github.com/aandvalenzuela/cmsdist/blob/a99b9b044d3ca31f8e5c3e3977c6c0c5414b3a53/llvm.spec#L8), [‎`openmpi`](https://github.com/aandvalenzuela/cmsdist/blob/a99b9b044d3ca31f8e5c3e3977c6c0c5414b3a53/openmpi.spec#L11), [`cuda-compatible-runtime`](https://github.com/aandvalenzuela/cmsdist/blob/a99b9b044d3ca31f8e5c3e3977c6c0c5414b3a53/cuda-compatible-runtime.spec#L8) and [`cuda`](https://github.com/aandvalenzuela/cmsdist/blob/a99b9b044d3ca31f8e5c3e3977c6c0c5414b3a53/cuda.spec#L17).

### CODE REVIEW

**Comment by aandvalenzuela** [comment]
test parameters:

- full_cmssw = true

**Comment by aandvalenzuela** [comment]
please test

- Let's test first with our regular image and then with the buildtime image

**Comment by smuzaffar** [comment]
please test

no need to run full_cmssw (change is only to check if oracle package deps are available in the platform seed or not)

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
@smuzaffar those libraries are provided by `glibc` that is present in the driver and the container image (runtime).  In this case we should move it to the buildtime image, right? Even if we move it to the buildtime image, do we need to add those `*.so` libraries as provides? I thought they should be already there by `glibc`.

**Comment by aandvalenzuela** [comment]
please test

File: cuda.spec [hunk 1]
```diff
@@ -14,7 +14,13 @@ Source0: https://developer.download.nvidia.com/compute/cuda/%{realversion}/local
...
 %endif
 Requires: python3
-AutoReq: no
+
+# FIXME: Needed by cuda when using default AutoReq. Provided by glibc, but not by glibc in our image.
+Provides: ld-linux.so.2
+Provides: ld-linux.so.2(GLIBC_2.3)
+Provides: libc.so.6(GLIBC_2.0)
+Provides: libc.so.6(GLIBC_2.1)
+Provides: libc.so.6(GLIBC_2.1.3)
```
Comment (smuzaffar) [inline cuda.spec]: looks like there are many system deps for cuda. I would suggest to keep `AutoReq: no` for cuda

**Comment by aandvalenzuela** [comment]
please test

**Comment by smuzaffar** [comment]
please test for el8_aarch64_gcc11

**Comment by smuzaffar** [comment]
please test for el8_ppc64le_gcc11

**Comment by aandvalenzuela** [comment]
please test for el8_ppc64le_gcc11

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test for slc7_amd64_gcc11

**Comment by smuzaffar** [comment]
please test for el8_aarch64_gcc11

**Comment by smuzaffar** [comment]
please test for el8_ppc64le_gcc11

**Comment by smuzaffar** [comment]
please test for el9_amd64_gcc11

**Pull request merged**